gammapy/visualization/utils.py

```diff
@@ -280,3 +286,111 @@ def plot_theta_squared_table(table):
     ax1.errorbar(x, table["sqrt_ts"], xerr=xerr, linestyle="None")
     ax1.set_xlabel(f"Theta [{theta2_axis.unit}]")
     ax1.set_ylabel("Significance")
+
+
+def plot_distribution(map_, ax=None, ncols=3, fit=True, fit_function="norm", **kwargs):
+    """
+    Plot the 1D distribution of data inside a map as a histogram. If the dimension of the map is smaller than 2,
+    a unique plot will be displayed. Otherwise, if the dimension is 3 or greater, a grid of plot will be displayed.
+
+    Parameters
+    ----------
+    map_ : an instance of `~gammapy.maps.Map`
+        A map that contains data to be plotted.
+    ax : `~matplotlib.axes.Axes` or list of `~matplotlib.axes.Axes`
+        Axis object to plot on. If a list of Axis is provided it has to be the same length as the length of _map.data.
+    ncols : int
+        Number of columns to plot if a "plot grid" was to be done.
+    fit : bool
+        Whether to perform a fit of the distribution of data. If True, the fit is performed by `scipy.stats`.
+    fit_function : str
+        A string of a name of a scipy continuous distribution from `scipy.stats._continuous_distns`.
+    **kwargs : dict
+        Keyword arguments to pass to `matplotlib.pyplot.hist` and `matplotlib.axes.Axes`.
+
+    Returns
+    -------
+    fit_parameters : list of tuple
+        List of tuple of the fitted parameters corresponding to the chosen `fit_function`. If `fit` is set to False,
+        returns an empty list.
+    axes : `~numpy.ndarray` of `~matplotlib.pyplot.Axes`
+        Array of Axes.
+    """
+
+    if not isinstance(map_, Map):
+        raise TypeError(
+            f"map_ must be an instance of gammapy.maps.Map, given {type(map_)}"
+        )
+
+    hist_args = list(inspect.signature(plt.hist).parameters)
```

Comment:
This is a very interesting approach of separating the keyword arguments. Right now I'm not sure if this really works, if there are non-unique parameters it must fail I guess. 

In Gammapy we typically have have approach to introduce different `kwargs` arguments like, `kwargs_hist`, `kwargs_axes`, `kwargs_patch`, this is slightly more inconvenient for the users, but simpler, easier to document and guaranteed to work.

Reply:
Yes I have tested it so it is working here ! I have checked that there are no shared parameters between these two objects. But yes it's probably not working if there are common parameters. 

I should probably stick to Gammapy way of handeling `kwargs` ! 

(I working on this function now !)